src/Semaphore.sol

```diff
@@ -271,11 +279,12 @@ contract Semaphore is IWorldID, SemaphoreCore {
     ///
     /// @param newRoot The new root to make the current root.
     function setCurrentRoot(uint256 newRoot) public onlyManager {
+        uint256 oldRoot = latestRoot;
         latestRoot = newRoot;
 
-        uint128 timestamp = rootHistory[newRoot];
+        uint128 timestamp = rootHistory[oldRoot];
```

Comment:
Can this ever not be zero?

Reply:
Yes. It gets set when a root gets added to the history, and hence has the value of `block.timestamp` at that point.